File: src/main/java/com/lucasgalmeida/llama/services/authservice/AuthService.java
package com.lucasgalmeida.llama.services.authservice;

import com.lucasgalmeida.llama.dto.auth.AuthResponseDTO;
import com.lucasgalmeida.llama.dto.auth.LoginRequestDTO;
import com.lucasgalmeida.llama.dto.auth.RegisterRequestDTO;

public interface AuthService {
    AuthResponseDTO login(LoginRequestDTO body);
    AuthResponseDTO register(RegisterRequestDTO body);
}

File: Frontend/src/services/auth.service.ts
import { LoginRequestDTO } from '../types/LoginRequestDTO';
import { RegisterRequestDTO } from '../types/RegisterRequestDTO';
import backend from './server-config';

const route = "/auth";

class AuthService {
    static async login(request:LoginRequestDTO) {
        try {
            const response = await backend.post(route + "/login", request);
            return response.data;
        } catch (error) {
            console.error(error);
            throw error;
        }
    }
    static async register(request:RegisterRequestDTO) {
        try {
            const response = await backend.post(route + "/register", request);
            return response.data;
        } catch (error) {
            console.error(error);
            throw error;
        }
    }
}

export default AuthService;
